services/object/src/backends/aws.js

```diff
@@ -0,0 +1,75 @@
+const { Backend } = require('./base');
+const assert = require('assert');
+const aws = require('aws-sdk');
+
+class AwsBackend extends Backend {
+  constructor(options) {
+    super(options);
+
+    for (let prop of ['accessKeyId', 'secretAccessKey', 'bucket', 'signGetUrls']) {
+      assert(prop in options.config, `backend ${options.backendId} is missing ${prop}`);
+    }
+
+    this.config = options.config;
+  }
+
+  async setup() {
+    const credentials = {
+      accessKeyId: this.config.accessKeyId,
+      secretAccessKey: this.config.secretAccessKey,
+    };
+    this.region = await getBucketRegion({ bucket: this.config.bucket, credentials });
+    this.s3 = new aws.S3({ region: this.region, ...credentials });
```

Comment:
Just a question that this brings to mind (not blocking landing this) how does this interact with being able to roll aws creds?

Reply:
I think just configuring new creds and restarting the service would do the trick.  But it's a good point regarding expiration of links, if we use years-long expirations..